src/pages/signup-affiliates/validations/account-details.tsx

```diff
@@ -0,0 +1,158 @@
+        return length_error
+    }
+
+    return null
+}
+const phoneValidation = (input, field_name, min_digit, max_digit) => {
+    if (!input) {
+        return <Localize translate_text="{{field_name}} is required" values={{ field_name }} />
+    } else if (
+        !validation_is_exceed_number(input, max_digit) ||
+        !validation_is_lack_number(input, min_digit)
+    ) {
+        return localize(`You should enter ${min_digit}-${max_digit} numbers.`)
+    } else if (!affiliate_validation_regex.phone.test(input)) {
+        return localize(`Please enter a valid phone number (e.g. +15417541234)`)
+    }
+    return null
+}
+
+const passwordValidation = (input, field_name, min_digit, max_digit) => {
+    if (!input) {
+        return <Localize translate_text="{{field_name}} is required" values={{ field_name }} />
+    } else if (
+        !validation_is_exceed_number(input, max_digit) ||
+        !validation_is_lack_number(input, min_digit)
+    ) {
+        return localize(`You should enter ${min_digit}-${max_digit} characters.`)
+    } else if (!affiliate_validation_regex.password.test(input)) {
+        return localize(`Password should have lower and uppercase English letters with numbers.`)
+    }
+    return null
+}
+const postcodeValidation = (input, field_name, min_digit, max_digit) => {
+    if (!input) {
+        return <Localize translate_text="{{field_name}} is required" values={{ field_name }} />
+    } else if (
+        !validation_is_exceed_number(input, max_digit) ||
+        !validation_is_lack_number(input, min_digit)
+    ) {
+        return localize(`You should enter ${min_digit}-${max_digit} characters.`)
+    } else if (affiliate_validation_regex.postal_code.test(input)) {
+        return localize(`Postcode is invalid`)
+    }
+}
+const stateValidation = (input, field_name, min_digit, max_digit) => {
+    if (!input) {
+        return <Localize translate_text="{{field_name}} is required" values={{ field_name }} />
+    } else if (
+        !validation_is_exceed_number(input, max_digit) ||
+        !validation_is_lack_number(input, min_digit)
+    ) {
+        return localize(`You should enter ${min_digit}-${max_digit} characters.`)
+    } else if (affiliate_validation_regex.state.test(input)) {
+        return localize(`Please enter a valid state`)
+    }
+}
+const cityValidation = (input, field_name, min_digit, max_digit) => {
+    if (!input) {
+        return <Localize translate_text="{{field_name}} is required" values={{ field_name }} />
+    } else if (
+        !validation_is_exceed_number(input, max_digit) ||
+        !validation_is_lack_number(input, min_digit)
+    ) {
+        return localize(`You should enter ${min_digit}-${max_digit} characters.`)
+    } else if (affiliate_validation_regex.city.test(input)) {
+        return localize(`Please enter a valid city`)
+    }
+}
+const validation = {
+    first_name: (input) => {
+        return nameValidation(input, localize('First Name'), 2, 50)
+    },
```

Comment:
We can declare outside a function with main validation logic and customize unique regex input inside `validation`
``` suggestion 
const Validate = (input, field_name, min_digit, max_digit, value="characters") => {
 if (!input) {
            return <Localize translate_text="{{field_name}} is required" values={{ field_name }} />
        } else if (
            !validation_is_exceed_number(input, max_digit) ||
            !validation_is_lack_number(input, min_digit)
        ) {
            return localize(`You should enter ${min_digit}-${max_digit} {value}.`)
        }
}
...
first_name: (input) => {
        return (Validate(input, localize('First Name'), 2, 50) || (affiliate_validation_regex.city.test(input)) {
        return localize(`Please enter a valid city`)
    })},
```

Or something closely like that for all validation type